[R] "keyword is hidden by macro definition" warnings with clang on CRAN

```
Version: 0.6.0
Check: whether package can be installed
Result: WARN
  Found the following significant warnings:
    ../src/flatcc/portable/pstdalign.h:144:9: warning: keyword is hidden by macro definition [-Wkeyword-macro]
    ../src/flatcc/portable/pstdalign.h:148:9: warning: keyword is hidden by macro definition [-Wkeyword-macro]
  See ‘/home/hornik/tmp/R.check/r-devel-clang/Work/PKGS/nanoarrow.Rcheck/00install.out’ for details.
  * used C compiler: ‘Debian clang version 19.1.7 (3)’
  * used C++ compiler: ‘Debian clang version 19.1.7 (3)’
Flavor: [r-devel-linux-x86_64-debian-clang](https://www.r-project.org/nosvn/R.check/r-devel-linux-x86_64-debian-clang/nanoarrow-00check.html)
```

```
Version: 0.6.0
Check: whether package can be installed
Result: WARN
  Found the following significant warnings:
    ../src/flatcc/portable/pstdalign.h:144:9: warning: keyword is hidden by macro definition [-Wkeyword-macro]
    ../src/flatcc/portable/pstdalign.h:148:9: warning: keyword is hidden by macro definition [-Wkeyword-macro]
  See ‘/data/gannet/ripley/R/packages/tests-clang/nanoarrow.Rcheck/00install.out’ for details.
  * used C compiler: ‘clang version 20.1.2’
  * used C++ compiler: ‘clang version 20.1.2’
Flavor: [r-devel-linux-x86_64-fedora-clang](https://www.r-project.org/nosvn/R.check/r-devel-linux-x86_64-fedora-clang/nanoarrow-00check.html)
```

<https://CRAN.R-project.org/package=nanoarrow> says "issues need fixing before 2025-04-16"

--- a/thirdparty/flatcc/include/flatcc/portable/pstdalign.h
+++ b/thirdparty/flatcc/include/flatcc/portable/pstdalign.h
@@ -140,6 +140,9 @@ extern "C" {
 
 #endif /* __STDC__ */
 
+// For C23, alignas/alignof are keywords and will warn (-Wkeyword-macro) when #defined here
+#if !(defined(__STDC__) && __STDC__ && defined(__STDC_VERSION__) && __STDC_VERSION__ >= 202311L)
+
 #ifndef alignas
 #define alignas _Alignas
 #endif
@@ -150,6 +153,8 @@ extern "C" {
 
 #define __alignas_is_defined 1
 #define __alignof_is_defined 1
+
+#endif
 
 #ifdef __cplusplus
 }

--- a/thirdparty/flatcc/include/flatcc/portable/pstdalign.h
+++ b/thirdparty/flatcc/include/flatcc/portable/pstdalign.h
@@ -151,10 +151,10 @@ extern "C" {
 #define alignof _Alignof
 #endif
 
+#endif
+
 #define __alignas_is_defined 1
 #define __alignof_is_defined 1
-
-#endif
 
 #ifdef __cplusplus
 }

